lib/wanda_web/controllers/executions_controller.ex

```diff
@@ -0,0 +1,34 @@
+defmodule WandaWeb.ExecutionsController do
+  use WandaWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Wanda.Results
+  alias WandaWeb.Schemas.CheckExecutionResponse
+
+  plug OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true
+
+  operation :list_checks_executions,
+    summary: "List check executions",
+    parameters: [
+      group_id: [
+        in: :query,
+        description: "Group ID to filter upon",
+        type: :string,
+        example: "00000000-0000-0000-0000-000000000001"
+      ],
+      page: [in: :query, description: "Page", type: :integer, example: 3],
+      items_per_page: [in: :query, description: "Items per page", type: :integer, example: 20]
+    ],
+    responses: [
+      ok: {"Check executions response", "application/json", CheckExecutionResponse}
+    ]
+
+  def list_checks_executions(conn, params) do
+    page = Map.get(params, :page, 1)
+    items_per_page = Map.get(params, :items_per_page, 10)
+    group_id = Map.get(params, :group_id)
```

Comment:
what happens if the user does not send any group_id?

We should straight match the group_id in the function header and then use this to generate an error in case the controller does not match:
https://hexdocs.pm/phoenix/controllers.html#action-fallback

Reply:
I acted under the assumption that if the user doesn't filter by group_id, the backend should just return unfiltered results. Do we wanna treat group_id as mandatory?

Reply:
No, it shouldn't be mandatory. It is a query param. If it doesn't come, the `group_id` we return all the executions. Actually, I like the current implementation on this part.

Reply:
ah damn sorry, my bad, it is not mandatory